include/grpcpp/security/tls_credentials_options.h

```diff
@@ -0,0 +1,260 @@
+/*
+ *
+ * Copyright 2019 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#ifndef GRPCPP_SECURITY_TLS_CREDENTIALS_OPTIONS_H
+#define GRPCPP_SECURITY_TLS_CREDENTIALS_OPTIONS_H
+
+#include <memory>
+#include <vector>
+
+#include <grpc/grpc_security.h>
+#include <grpc/support/log.h>
+#include <grpcpp/support/config.h>
+
+namespace grpc_impl {
+namespace experimental {
+
+/** TLS key materials config, wrapper for grpc_tls_key_materials_config. **/
+class TlsKeyMaterialsConfig {
+ public:
+  struct PemKeyCertPair {
+    grpc::string private_key;
+    grpc::string cert_chain;
+  };
+
+  /** Getters for member fields. **/
+  const grpc::string pem_root_certs() const { return pem_root_certs_; }
+  const std::vector<PemKeyCertPair>& pem_key_cert_pair_list() const {
+    return pem_key_cert_pair_list_;
+  }
+  int version() const { return version_; }
+
+  /** Setter for key materials that will be called by the user. The setter
+   * transfers ownership of the arguments to the config. **/
+  void set_key_materials(grpc::string pem_root_certs,
+                         std::vector<PemKeyCertPair> pem_key_cert_pair_list);
+  void set_version(int version) { version_ = version; };
+
+ private:
+  int version_;
+  std::vector<PemKeyCertPair> pem_key_cert_pair_list_;
+  grpc::string pem_root_certs_;
+};
+
+/** The following 2 functions are exposed for testing purposes. **/
+grpc_tls_key_materials_config* c_key_materials(
```

Comment:
Could you please use C++ naming style for these exposed functions, e.g., ConvertToCKeyMaterialsConfig()? - "functions should start with a capital letter and have a capital letter for each new word." 

I also noticed there are other functions exposed for testing in this file, and how about placing them all in a new file - tls_credentials_options_util.{h, cc}? which I think it will be much cleaner. After all they are internal implementation details, and should not be exposed to the users.

Reply:
I've changed the naming for these two functions. However, for some other of the exposed functions (e.g. "tls_credential_reload_config_c_schedule"), I have put them in the util file, but left the C-style naming. The reason for this is that these are functions called by the C-core API's (in this case, the C version of the user-provided schedule function). Does that work? Or would you prefer these also be named as C++-style functions?